backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -95,6 +95,32 @@
         .collect(Collectors.toList());
   }
 
+  private Optional<Engagement> fromDto(EngagementDto engagement) {
```

Comment:
Would expect mapping function as part of EngagementDto class (at least this is recommended approach in C#)

Reply:
The slight complication here is that EngagementDto depends on StaffId, while Engagement depends directly on Staff object derived from the staffId, which needs to be fetched from the database (using staffRepository object).
I didn't want to introduce database dependencies to the DTO, so I put it here. The alternative is to inject staffRepository to another class, which I can do if you think it's better.